python/lib/client/dmod/client/__init__.py

```diff
@@ -1 +1,22 @@
-name = 'client'
\ No newline at end of file
+name = 'client'
+
+from dmod.core.data_domain_detectors import ItemDataDomainDetectorRegistry
+
+
+def register_modeldata_domain_detectors() -> bool:
+    """
+    Register the known item domain detectors implemented within the ``dmod.modeldata`` package, if it is installed.
+
+    Returns
+    -------
+    bool
+        Whether the registration succeeded, which typically equates to whether ``dmod.modeldata`` was installed.
+    """
+    try:
+        from dmod.modeldata.data.item_domain_detector import (AorcCsvFileDomainDetector, RealizationConfigDomainDetector,
+                                                              GeoPackageHydrofabricDomainDetector)
+        for d in (AorcCsvFileDomainDetector, GeoPackageHydrofabricDomainDetector, RealizationConfigDomainDetector):
```

Comment:
That's what the return value is for. I envisioned something very (and perhaps overly) flexible. To not assume that any particular source of a detector was available. To only care whether there was or wasn't _something_ that was sufficient. That's part of why I didn't make `modeldataa` dependency here. So it's fine if that doesn't work; we really don't even need the indication of success, and returning one allows the caller to react appropriately. 

All that said, I am rethinking things some at the moment. I wanted the capability to introduce detectors easily that might be developed externally, but how likely is that really?

Reply:
In either case, for better or worse, its not particularly pythonic to use a boolean in this way. What I mean is, I would have to read the documentation / the source if I were to run into an issue with data detectors not working when I thought that they should. I agree that, ideally you would just return an exception by value instead of raising thing (I personally am not a fan of raising exceptions, I would much prefer they were just values). Then you can ignore it if that makes sense for your use case.

Reply:
We can ditch the boolean return value:  it isn't actually being check.  I see your point about the documentation, although I had hoped the function to print what detectors were being loaded would be sufficient compensation.  

The primary goal I'd still like to achieve here is basically loading from something that isn't strictly a dependency.   Is there a better way around the ambiguity problem you mention, other than the aforementioned introspection into what detectors were loaded, beyond just abandoning that goal?

Reply:
For now, for the sake of simplicity, I'm abandoning the aforementioned dependency avoidance goal.